subsys/usb/device_next/class/usbd_cdc_acm.c

```diff
@@ -360,14 +360,14 @@ static void usbd_cdc_acm_enable(struct usbd_class_data *const c_data)
 		cdc_acm_irq_rx_enable(dev);
 	}
 
-	if (atomic_test_bit(&data->state, CDC_ACM_IRQ_TX_ENABLED)) {
-		if (ring_buf_space_get(data->tx_fifo.rb)) {
+	if (ring_buf_space_get(data->tx_fifo.rb)) {
```

Comment:
I think the observed "infinite loop" is not relevant here and is rather confusing. When CDC ACM instance gets enabled and there is some data in TX FIFO, the TX FIFO work will not be triggered if no UART API is used. And if TX FIFO was full before CDC ACM instance get enabled, TX work will never be triggered. To fix it TX FIFO should always be drained when CDC ACM gets enabled.

```suggestion
	if (ring_buf_is_empty(data->tx_fifo.rb)) {
```